### PR TITLE
crud operations tracked and logged

### PR DESCRIPTION
**Pull request by SunmadeBabafemi**
@Chimexx  kindly try the migration stuff again

### CODE REVIEW

File: src/database/migration/1664894757627-crudLogger.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE "cat_entity" DROP CONSTRAINT "FK_26db7bdfb66a9c72a48e4fa2854"`);
+        await queryRunner.query(`ALTER TABLE "cat_toy_entity" DROP CONSTRAINT "FK_441e45eec97723bb12d5123213a"`);
+        await queryRunner.query(`DROP TABLE "logger"`);
+        await queryRunner.query(`DROP TYPE "public"."logger_action_type_enum"`);
+        await queryRunner.query(`DROP TABLE "cat_home_entity"`);
+        await queryRunner.query(`DROP TABLE "cat_entity"`);
+        await queryRunner.query(`DROP TABLE "cat_toy_entity"`);
```
Comment (Chimexx) [inline src/database/migration/1664894757627-crudLogger.ts:23]: This is not right

**Review by Chimexx — CHANGES_REQUESTED** [review]

File: src/database/migration/1664894757627-crudLogger.ts (hunk 1 above)
Comment (SunmadeBabafemi) [reply src/database/migration/1664894757627-crudLogger.ts:23]: i dont understand... i think someone did something that wasnt tracked with a migration file.. so anytime i generate migration for my own entities those queries also generate with it...

Comment (Chimexx) [reply src/database/migration/1664894757627-crudLogger.ts:23]: Hmmm 🥴🥴

Comment (SunmadeBabafemi) [reply src/database/migration/1664894757627-crudLogger.ts:23]: but run it locally if it works fine on yours

Comment (johnayinde) [reply src/database/migration/1664894757627-crudLogger.ts:23]: I wondered  where those Tables are coming from, also encountered same thing while generating migration 
I had to cut the lines off before running

File: src/common/interfaces/loggerJson.interface.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { Visibility } from 'src/common/enums/visibility.enum';
+import { ReqBody } from 'src/endpoints/interface/endpoint.interface';
+import { HttpMethod } from '../enums/httpMethods.enum';
+
+
+export type LoggerJson = {
```
Comment (PhilemonBrain) [inline src/common/interfaces/loggerJson.interface.ts:6]: @SunmadeBabafemi hope these required values wont throw an error sometime in d future

File: src/logger/logger.controller.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  Post,
+} from '@nestjs/common';
+import { LoggerService } from './logger.service';
+import { Ok, ZaLaResponse } from '../common/helpers/response';
+import { ApiOperation, ApiTags } from '@nestjs/swagger';
+import { Logger } from 'src/entities/logger.entity';
+import { SearchLogDto } from './dto/search-log.dto';
+import { Paginate, Paginated, PaginateQuery } from 'nestjs-paginate';
+
+@ApiTags('CRUD Logs')
```
Comment (PhilemonBrain) [inline src/logger/logger.controller.ts:15]: can u rename to Activity Logs

File: src/database/migration/1664894757627-crudLogger.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class crudLogger1664894757627 implements MigrationInterface {
+    name = 'crudLogger1664894757627'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "cat_toy_entity" ("id" SERIAL NOT NULL, "name" character varying NOT NULL, "createdAt" TIMESTAMP NOT NULL DEFAULT now(), "catId" integer, CONSTRAINT "PK_ac5c1672d60b151112f65ebcf09" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`CREATE TABLE "cat_entity" ("id" SERIAL NOT NULL, "name" character varying NOT NULL, "color" character varying NOT NULL, "age" integer, "createdAt" TIMESTAMP NOT NULL DEFAULT now(), "deletedAt" TIMESTAMP, "homeId" integer, CONSTRAINT "REL_26db7bdfb66a9c72a48e4fa285" UNIQUE ("homeId"), CONSTRAINT "PK_676cf6dbf9d1d7d216ecd87c4f3" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`CREATE TABLE "cat_home_entity" ("id" SERIAL NOT NULL, "name" character varying NOT NULL, "createdAt" TIMESTAMP NOT NULL DEFAULT now(), CONSTRAINT "PK_6bdfe5785861b7dad300f8f0ec0" PRIMARY KEY ("id"))`);
```
Comment (PhilemonBrain) [inline src/database/migration/1664894757627-crudLogger.ts:9]: please can u also remove these cat entity values from the migration file??

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]

**Comment by PhilemonBrain** [comment]
@SunmadeBabafemi Nice work! thanks. please check the requested changes

File: src/common/interfaces/loggerJson.interface.ts (hunk 2 above)
Comment (SunmadeBabafemi) [reply src/common/interfaces/loggerJson.interface.ts]: not at all.. there are not required perse... they are fields that the frontend can use to populate changed fields

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**